sdk/src/main/java/io/opentelemetry/sdk/trace/export/SpanData.java

```diff
@@ -0,0 +1,544 @@
+        .status(status)
+        .endTimestamp(endTimestamp)
+        .build();
+  }
+
+  /**
+   * Returns the {@code SpanContext} associated with this {@code Span}.
+   *
+   * @return the {@code SpanContext} associated with this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract SpanContext getContext();
+
+  /**
+   * Returns the parent {@code SpanId}. If the {@code Span} is a root {@code Span}, the SpanId
+   * returned will be invalid..
+   *
+   * @return the parent {@code SpanId} or an invalid SpanId if this is a root {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract SpanId getParentSpanId();
+
+  /**
+   * Returns the resource of this {@code Span}.
+   *
+   * @return the resource of this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract Resource getResource();
+
+  /**
+   * Returns the name of this {@code Span}.
+   *
+   * @return the name of this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract String getName();
+
+  /**
+   * Returns the kind of this {@code Span}.
+   *
+   * @return the kind of this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract Kind getKind();
+
+  /**
+   * Returns the start {@code Timestamp} of this {@code Span}.
+   *
+   * @return the start {@code Timestamp} of this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract Timestamp getStartTimestamp();
+
+  /**
+   * Returns the attributes recorded for this {@code Span}.
+   *
+   * @return the attributes recorded for this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract Map<String, AttributeValue> getAttributes();
+
+  /**
+   * Returns the timed events recorded for this {@code Span}.
+   *
+   * @return the timed events recorded for this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract List<TimedEvent> getTimedEvents();
+
+  /**
+   * Returns links recorded for this {@code Span}.
+   *
+   * @return links recorded for this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract List<io.opentelemetry.trace.Link> getLinks();
+
+  /**
+   * Returns the {@code Status}.
+   *
+   * @return the {@code Status}.
+   * @since 0.1.0
+   */
+  public abstract Status getStatus();
+
+  /**
+   * Returns the end {@code Timestamp}.
+   *
+   * @return the end {@code Timestamp}.
+   * @since 0.1.0
+   */
+  public abstract Timestamp getEndTimestamp();
+
+  /**
+   * The trace id for this span, retrieved from the associated SpanContext.
+   *
+   * @return the TraceId
+   */
+  public TraceId getTraceId() {
+    return getContext().getTraceId();
+  }
+
+  /**
+   * The id of this span, retrieved from the associated SpanContext.
+   *
+   * @return the SpanId
+   */
+  public SpanId getSpanId() {
+    return getContext().getSpanId();
+  }
+
+  /**
+   * An immutable implementation of {@link io.opentelemetry.trace.Link}.
+   *
+   * @since 0.1.0
+   */
+  @Immutable
+  @AutoValue
+  public abstract static class Link implements io.opentelemetry.trace.Link {
```

Comment:
This is not needed, use from the API `util.Links.create`

Reply:
Agreed. I had been planning on cleaning all of this up after the fact, but we can push that up front.  (we had just copied @carlosalberto 's `SpanData` class as the starting point)